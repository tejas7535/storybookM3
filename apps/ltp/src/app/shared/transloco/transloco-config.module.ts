import { APP_INITIALIZER, ModuleWithProviders, NgModule } from '@angular/core';

import {
  getBrowserLang,
  HashMap,
  TRANSLOCO_CONFIG,
  TranslocoConfig,
  TranslocoModule,
  TranslocoService
} from '@ngneat/transloco';

import { environment } from '../../../environments/environment';
import { translocoLoader } from './transloco.loader';

const translocoConfig: TranslocoConfig = {
  availableLangs: ['en', 'de'],
  defaultLang: 'en',
  fallbackLang: 'en',
  prodMode: environment.production,
  reRenderOnLangChange: true,
  flatten: {
    aot: environment.production
  }
};

const preloadLanguage = (transloco: TranslocoService): Function => {
  return async (): Promise<HashMap> => {
    const lang = getBrowserLang();

    transloco.setActiveLang(lang);

    return transloco.load(lang).toPromise();
  };
};

const preLoad = {
  provide: APP_INITIALIZER,
  multi: true,
  useFactory: preloadLanguage,
  deps: [TranslocoService]
};

@NgModule({
  imports: [TranslocoModule],
  exports: [TranslocoModule]
})
export class TranslocoConfigModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: TranslocoConfigModule,
      providers: [
        {
          provide: TRANSLOCO_CONFIG,
          useValue: translocoConfig
        },
        translocoLoader,
        preLoad
      ]
    };
  }
}
