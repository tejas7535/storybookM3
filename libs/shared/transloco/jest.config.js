module.exports = {
  name: 'shared-transloco',
  preset: '../../../jest.config.js',
  coverageDirectory: '../../../coverage/libs/shared/transloco',
  snapshotSerializers: [
    'jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js',
    'jest-preset-angular/build/AngularSnapshotSerializer.js',
    'jest-preset-angular/build/HTMLCommentSerializer.js'
  ]
};
