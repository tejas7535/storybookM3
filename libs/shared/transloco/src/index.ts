export * from './lib/shared-transloco.module';
export * from './lib/shared-transloco-testing.module';
export * from './lib/shared-scope.loader';
