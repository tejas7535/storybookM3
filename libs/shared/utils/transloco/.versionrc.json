{
  "extends": "../../../../.versionrc.json",
  "path": ".",
  "releaseCommitMessageFormat": "chore(transloco): ðŸš€  {{currentTag}}  [ci skip]",
  "tagPrefix": "transloco-v",
  "skip": {
    "bump": false,
    "changelog": false,
    "commit": false,
    "tag": false
  }
}
