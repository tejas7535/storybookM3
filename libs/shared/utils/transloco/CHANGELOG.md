 Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.
**Note:** Dependency updates, refactored code & style/test/performance changes are not shown within this changelog. Thus, releases without any entries may occur.

### [3.0.1](https://github.com/Schaeffler-Group/frontend-schaeffler/compare/transloco-v3.0.1...transloco-v3.0.0) (2021-09-07)

## [3.0.0](https://github.com/Schaeffler-Group/frontend-schaeffler/compare/transloco-v3.0.0...transloco-v2.5.0) (2021-09-07)


### ⚠ BREAKING CHANGES

* **transloco:** update peer dependencies

### 🎸 Features

* **libs:** adjust rxjs and ngrx versions ([f6c92d8](https://github.com/Schaeffler-Group/frontend-schaeffler/commit/f6c92d81ace947127362bd322283a8ac925ab998))
* **libs:** use partial compilation (UFTABI-4907) ([#2835](https://github.com/Schaeffler-Group/frontend-schaeffler/issues/2835)) ([27829ff](https://github.com/Schaeffler-Group/frontend-schaeffler/commit/27829ff96da6ccc3a4ee0b98bc6f766a8c4a5057))
* **transloco:** update peer dependencies ([3646d5c](https://github.com/Schaeffler-Group/frontend-schaeffler/commit/3646d5c91091e3683a1d588a38fb5faa07e9dfe9))

## [2.5.0](https://github.com/Schaeffler-Group/frontend-schaeffler/compare/transloco-v2.5.0...transloco-v2.4.1) (2021-08-24)


### 🎸 Features

* **workspace:** fix accessibility and numerical separator issues (UFTABI-4728) ([699fb97](https://github.com/Schaeffler-Group/frontend-schaeffler/commit/699fb97a63a9069d847dfa489386da561028e5ea))


### 🐛 Bug Fixes

* **storybook:** fix transloco usage ([#2705](https://github.com/Schaeffler-Group/frontend-schaeffler/issues/2705)) ([463eb94](https://github.com/Schaeffler-Group/frontend-schaeffler/commit/463eb9486fca3c66236f4e0a1b38c4c52d8e5f9a))

### [2.4.1](///compare/transloco-v2.4.1...transloco-v2.4.0) (2021-05-20)

## [2.4.0](///compare/transloco-v2.4.0...transloco-v2.3.0) (2021-04-29)


### 🎸 Features

* **mm:** add manual switch for language and decimal separator ([291504b](///commit/291504b1a4bb92e896514dc2d9f033d7efc67bab))
* **transloco:** update transloco testing module (UFTABI-4323) ([47630c6](///commit/47630c62ca451d70e613182684fc34506a34705a))

## [2.3.0](///compare/transloco-v2.3.0...transloco-v2.1.0) (2021-03-24)


### 🎸 Features

* **transloco:** support cache busting (DSCDA-1780) ([b3977ae](///commit/b3977aefd6534d8be3689f2240bac6f4b5861fa8))
* **workspace:** build libs with prod config (UFTABI-4112) ([6cd84a2](///commit/6cd84a2b3f3b5fe695d93c28e6cf5eb69bf6c205))


### 🐛 Bug Fixes

* **workspace:** adjust libs version ([df88244](///commit/df88244a1a49ef9d4eef59a2e6b2e5cd5e2de976))

## [2.1.0](///compare/transloco-v2.1.0...transloco-v2.0.0) (2021-01-22)


### 🎸 Features

* **deps:** update to nx 11 and fix jest setup ([4df2df3](///commit/4df2df38f8a3fa29abae9b9f736e7d237344541b))

## 2.0.0 (2020-11-25)


### ⚠ BREAKING CHANGES

* **header:** Header, Responsive and Transloco are moved to new publishable libraries (@schaeffler/shared/ui-components --> `@schaeffler/header`; @schaeffler/shared/responsive --> `@schaeffler/responsive`; @schaeffler/shared/transloco--> `@schaeffler/transloco`)

### 🎸 Features

* **libs:** use view engine and prevent barrel imports in banner and transloco (UFTABI-2733, UFTABI-2741) ([79024d5](///commit/79024d550448ec650a612566e85009158fb9788f))
* **workspace:** enable auto-populated publishable library dependencies (UFTABI-2390) ([471e246](///commit/471e246144837957500060590020b380a0940c39))
* release v1 (UFTABI-2483) ([79c1ba7](///commit/79c1ba7c6c1af8ccd909083d91fffbe0ae017ebb))
* **header:** split header into seperate publishable lib (UFTABI-2309) ([721ead8](///commit/721ead8681c9ce017e6ff939911dc31d449831f7))


### 📈 Improvements

* **workspace:** adjust libs to ensure angular 11 compatibility ([08e417a](///commit/08e417a2e3a8c2404681863ff6466216e9ba80c6))

### [1.3.5](///compare/v1.9.0...v1.3.5) (2020-11-02)

### [1.3.4](https://gitlab.schaeffler.com/frontend-schaeffler/schaeffler-frontend/compare/v1.9.0...v1.3.4) (2020-10-27)

### [1.3.3](///compare/v1.7.0...v1.3.3) (2020-10-12)

### [1.3.2](///compare/v1.6.0...v1.3.2) (2020-09-28)

### [1.3.1](///compare/v1.5.0...v1.3.1) (2020-09-11)

## [1.3.0](///compare/v1.4.0...v1.3.0) (2020-08-27)


### 🎸 Features

* **libs:** use view engine and prevent barrel imports in banner and transloco (UFTABI-2733, UFTABI-2741) ([79024d5](///commit/79024d550448ec650a612566e85009158fb9788f))

### [1.2.2](///compare/v1.3.0...v1.2.2) (2020-08-12)

### [1.2.1](///compare/v1.2.0...v1.2.1) (2020-08-07)

## [1.2.0](///compare/v1.1.0...v1.2.0) (2020-07-21)


### 🎸 Features

* **workspace:** enable auto-populated publishable library dependencies (UFTABI-2390) ([471e246](///commit/471e246144837957500060590020b380a0940c39))

## [1.1.0](///compare/v0.5.0...v1.1.0) (2020-07-01)


### 🎸 Features

* release v1 (UFTABI-2483) ([79c1ba7](///commit/79c1ba7c6c1af8ccd909083d91fffbe0ae017ebb))

## [1.0.0](///compare/v0.5.0...v1.0.0) (2020-06-02)

**⚠ First official release that contains all previous releases ⚠**