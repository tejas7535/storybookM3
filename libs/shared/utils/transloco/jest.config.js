module.exports = {
  name: 'shared-utils-transloco',
  preset: '../../../../jest.config.js',
  coverageDirectory: '../../../../coverage/libs/shared/utils/transloco',
  snapshotSerializers: [
    'jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js',
    'jest-preset-angular/build/AngularSnapshotSerializer.js',
    'jest-preset-angular/build/HTMLCommentSerializer.js',
  ],
};
