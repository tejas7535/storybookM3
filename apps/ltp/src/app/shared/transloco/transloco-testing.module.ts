import { TranslocoConfig, TranslocoTestingModule } from '@ngneat/transloco';

import * as en from '../../../assets/i18n/en.json';

export const getTranslocoModule = (config: Partial<TranslocoConfig> = {}) =>
  TranslocoTestingModule.withLangs(
    { en },
    {
      availableLangs: ['en'],
      defaultLang: 'en',
      ...config
    }
  );
