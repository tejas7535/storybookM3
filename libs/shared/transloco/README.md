# shared-transloco

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-transloco` to execute the unit tests.

## How to use transloco in your app

Import `SharedTranslocoModule` into your main app.module.ts:  
```
import { environment } from '../environments/environment';
.
.
.
imports: [
    SharedTranslocoModule.forRoot(environment.production, ['de', 'en'], true)
]

```

The first parameter represents whether Transloco should be imported for production with AOT. The second parameter represents the available languages in your application.  The third parameter defines whether the app itself has translation files or not.  

**important**: do provide an array with desired languages even though you not provides translations since libraries that have translations do need this information. Besides, you need this module when you have libraries in your app that have translations. If you do not have translations in your app but you have libraries with i18n you need to set the third parameter to false.

## How to use transloco in your library

Import `SharedTranslocoModule` into your module:
```
import { environment } from '../environments/environment';

const loader = ['en'].reduce((acc: any, lang: string) => {
  acc[lang] = () => import(`../assets/i18n/${lang}.json`);
  return acc;
}, {});
.
.
.
imports: [
    SharedTranslocoModule.forChild('scope', loader)
]

```

The first parameter represents the scope that is used within the library for translation.  
Do not forget to add `SharedTranslocoModule.forRoot` into your main app later.

## How to unit test translations

Please import `provideTranslocoTestingModule` into your .spec from `'@schaeffler/shared/transloco'`.  
`provideTranslocoTestingModule` expects the i18n json files as a parameter.

Here is an example:
```
import { provideTranslocoTestingModule } from '@schaeffler/shared/transloco';
import * as en from '../../../../assets/i18n/en.json';

configureTestSuite(() => {
  TestBed.configureTestingModule({
    ...
    imports: [provideTranslocoTestingModule({ en })],
    ...
  });
});

```

