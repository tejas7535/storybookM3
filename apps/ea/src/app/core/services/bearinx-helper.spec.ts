import { TranslocoModule } from '@jsverse/transloco';

import {
  extractErrorsFromResult,
  extractErrorsWarningsAndNotesFromResult,
  extractNotesFromResult,
  extractSubordinatesFromPath,
  extractTableFromSubordinate,
  extractWarningsFromResult,
  formatMessageSubordinates,
  matchItem,
} from './bearinx-helper';
import {
  STRING_ERROR_BLOCK,
  STRING_NOTE_BLOCK,
  STRING_WARNING_BLOCK,
} from './bearinx-result.constant';
import {
  BearinxOnlineResult,
  BearinxOnlineResultSubordinate,
} from './bearinx-result.interface';

jest.mock('@jsverse/transloco', () => ({
  ...jest.requireActual<TranslocoModule>('@jsverse/transloco'),
  translate: jest.fn((translateKey) => translateKey),
}));

describe('Bearinx Helper', () => {
  describe('matchItem', () => {
    it('should match a correct item correctly', () => {
      const item: BearinxOnlineResultSubordinate = {
        identifier: 'abc',
        subordinates: [],
        title: 'title',
      };

      expect(matchItem(item, { identifier: 'abc' })).toBe(true);
      expect(matchItem(item, { identifier: 'abc', title: 'title' })).toBe(true);
      expect(
        matchItem(
          { ...item, titleID: 'my-id' },
          { identifier: 'abc', title: 'title', titleID: 'my-id' }
        )
      ).toBe(true);
    });

    it('should not match an incorrect item', () => {
      const item: BearinxOnlineResultSubordinate = {
        identifier: 'abc',
        subordinates: [],
        title: 'title',
      };

      expect(matchItem(item, { identifier: 'abcd' })).toBe(false);
      expect(matchItem(item, { identifier: 'abc', title: 'title-2' })).toBe(
        false
      );
      expect(
        matchItem(
          { ...item, titleID: 'my-id' },
          { identifier: 'abc', title: 'title', titleID: 'my-id-2' }
        )
      ).toBe(false);
    });
  });

  describe('extractSubordinatesFromPath', () => {
    const needle: BearinxOnlineResultSubordinate = {
      identifier: 'abc',
      subordinates: [],
      title: 'my-title',
    };
    const input: BearinxOnlineResult = {
      subordinates: [
        { identifier: 'sub', subordinates: [needle], title: 'sub-title' },
      ],
    } as BearinxOnlineResult;

    it('returns the correct subordinate', () => {
      expect(
        extractSubordinatesFromPath(input, [
          { identifier: 'sub' },
          { identifier: 'abc' },
        ])
      ).toEqual(needle);
      expect(
        extractSubordinatesFromPath(input, [
          { title: 'sub-title' },
          { title: 'my-title' },
        ])
      ).toEqual(needle);
    });

    it('handles invalid paths gracefully', () => {
      expect(
        extractSubordinatesFromPath(input, [
          { identifier: 'sub-no' },
          { identifier: 'abc' },
        ])
      ).toBeUndefined();
    });
  });

  describe('extractTableFromSubordinate', () => {
    it('extracts data from a table', () => {
      const input: Partial<BearinxOnlineResultSubordinate> = {
        data: {
          fields: [],
          unitFields: [],
          items: [
            [
              { field: 'abc', unit: 'm', value: '1' },
              { field: 'def', unit: 'm', value: '2' },
              { field: 'resultValues.loadcase', value: 'loadcase' },
            ],
          ],
        },
      };

      expect(extractTableFromSubordinate(input)).toEqual([
        {
          abc: {
            unit: 'm',
            value: '1',
            short: 'abc',
            loadcaseName: 'loadcase',
          },
          def: {
            unit: 'm',
            value: '2',
            short: 'def',
            loadcaseName: 'loadcase',
          },
        },
      ]);
    });

    it('handles an empty input gracefully', () => {
      expect(extractTableFromSubordinate({})).toBeUndefined();
    });
  });

  describe('extractErrorsWarningsAndNotesFromResult', () => {
    let item: BearinxOnlineResultSubordinate;
    describe('when report messages are provided', () => {
      it('should extract report messages data', () => {
        item = {
          identifier: 'abc',
          subordinates: [
            {
              title: 'some title',
              identifier: 'block',
            },
          ],
        };
        expect(extractErrorsWarningsAndNotesFromResult(item)).toEqual([
          { title: 'some title', identifier: 'block' },
        ]);
      });
    });

    describe('when report messages are not provided', () => {
      it('should return empty collection', () => {
        item = {
          identifier: 'abc',
          subordinates: [
            {
              title: 'some title',
              identifier: 'block',
              titleID: 'some other result',
            },
          ],
        };
        expect(extractErrorsWarningsAndNotesFromResult(item)).toEqual([]);
      });
    });
  });

  describe('when getting errors, warnings and notes', () => {
    let items: BearinxOnlineResultSubordinate[];
    let result: BearinxOnlineResultSubordinate;

    beforeAll(() => {
      items = [
        {
          identifier: 'block',
          title: 'Warnings',
          titleID: STRING_WARNING_BLOCK,
          subordinates: [
            {
              text: ['some text'],
              identifier: 'text',
            },
          ],
        },
        {
          identifier: 'block',
          title: 'Errors',
          titleID: STRING_ERROR_BLOCK,
          subordinates: [
            {
              text: ['main error'],
              identifier: 'text',
            },
            {
              identifier: 'block',
              subordinates: [
                {
                  text: ['nested error 1'],
                  identifier: 'text',
                },
                {
                  text: [
                    'nested error 2',
                    'nested error 2.1',
                    'nested error 2.2',
                  ],
                  identifier: 'text',
                },
                {
                  text: ['nested error 3'],
                  identifier: 'text',
                },
              ],
            },
          ],
        },
        {
          identifier: 'block',
          title: 'Notes',
          titleID: STRING_NOTE_BLOCK,
          subordinates: [
            {
              text: ['some note'],
              identifier: 'text',
            },
            {
              identifier: 'block',
              subordinates: [
                {
                  text: ['note'],
                  identifier: 'text',
                },
                {
                  text: ['nested note 1', '    ', 'nested note 2', '    '],
                  identifier: 'text',
                },
                {
                  text: ['nested note 3'],
                  identifier: 'text',
                },
              ],
            },
          ],
        },
      ];

      result = {
        identifier: 'outputDescription',
        subordinates: items,
      };
    });

    describe('when getting errors from result', () => {
      let errors: BearinxOnlineResultSubordinate[];
      beforeAll(() => {
        errors = extractErrorsFromResult(result);
      });

      it('should extract errors', () => {
        expect(errors).toEqual([items[1]]);
      });

      describe('when formatting errors', () => {
        it('should format it', () => {
          expect(formatMessageSubordinates(errors)).toMatchSnapshot();
        });
      });
    });

    describe('when getting warnings from result', () => {
      let warnings: BearinxOnlineResultSubordinate[];
      beforeAll(() => {
        warnings = extractWarningsFromResult(result);
      });

      it('should extract warnings', () => {
        expect(warnings).toEqual([items[0]]);
      });

      describe('when formatting warnings', () => {
        it('should format it', () => {
          expect(formatMessageSubordinates(warnings)).toMatchSnapshot();
        });
      });
    });

    describe('when getting notes from result', () => {
      let notes: BearinxOnlineResultSubordinate[];
      beforeAll(() => {
        notes = extractNotesFromResult(result);
      });

      it('should extract notes', () => {
        expect(notes).toEqual([items[2]]);
      });

      describe('when formatting notes', () => {
        it('should format it', () => {
          expect(formatMessageSubordinates(notes)).toMatchSnapshot();
        });
      });
    });
  });
});
