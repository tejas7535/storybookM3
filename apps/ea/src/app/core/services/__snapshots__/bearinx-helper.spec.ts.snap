// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bearinx Helper when getting errors, warnings and notes when getting errors from result when formatting errors should format it 1`] = `
Array [
  "main error",
  "nested error 1",
  "nested error 2",
  "nested error 2.1",
  "nested error 2.2",
  "nested error 3",
]
`;

exports[`Bearinx Helper when getting errors, warnings and notes when getting notes from result when formatting notes should format it 1`] = `
Array [
  "some note",
  "note",
  "nested note 1",
  "nested note 2",
  "nested note 3",
]
`;

exports[`Bearinx Helper when getting errors, warnings and notes when getting warnings from result when formatting warnings should format it 1`] = `
Array [
  "some text",
]
`;
